### DISCUSSION TITLE
How do you find the issuer_id for an App Store Connect API Key?

### DISCUSSION THREAD

**Discussion by axelson** [fastlane/fastlane#18926] [Q&A] [answered]
Hi, I'm trying to follow the guides here:
https://docs.fastlane.tools/app-store-connect-api/

and I've successfully created and downloaded the App Store Connect API key and then the docs state:

> The API Key (located in the .p8) file, the key id, and the issuer id are needed to create the JWT token for authorization. There are multiple ways that these pieces of information can be input into fastlane.

But it doesn't tell you how to find or generate the issuer id. The issuer id looks like a UUIDv4, am I supposed to generate one myself?

I need the issuer id to provide to game.ci: https://game.ci/docs/github/deployment/ios (I need to fill out these secrets: APPSTORE_KEY_ID, APPSTORE_ISSUER_ID, APPSTORE_P8)

**Comment by crazymanish** [comment, marked as answer]
Hi @axelson 

Steps to find your `issuer_id`
- Login to the AppStoreConnect portal
- Select the `Users and Access` tab
- Select the `Keys` tab
- Copy the `issuer_id`

Please see attached image, hope this will help!
.
<img width="1107" alt="Screenshot 2021-06-19 at 20 32 18" src="https://user-images.githubusercontent.com/5364500/122652200-cd269480-d13d-11eb-8391-0606f2edb477.png">

**Reply by axelson** [reply]
Ah, thanks! Not sure how I missed that initially. I guess I was heavily expecting the issuer id to be associated with each key instead of the whole organization account.

**Reply by danielgomezrico** [reply]
In the new user interface, the "Keys" tab has been replaced by the "Integrations" tab. May I know how to obtain this tab?



<img width="812" alt="image" src="https://github.com/fastlane/fastlane/assets/76348/5ccd56c6-9ee4-4fc7-9f71-55b5ea2a0e82">

**Reply by tk-nguyen** [reply]
@danielgomezrico The keys are in the `Integrations` tab now:

![image](https://github.com/fastlane/fastlane/assets/39090621/790c57ea-6cbb-4e26-a804-125f071d66a0)

**Reply by danielgomezrico** [reply]
Oh thanks! 

So if I generate a personal API key it cannot be used? it should be a team key?

**Reply by tk-nguyen** [reply]
> Oh thanks! 
> 
> So if I generate a personal API key it cannot be used? it should be a team key?

Yes, I think so. More info here: https://developer.apple.com/documentation/appstoreconnectapi/creating_api_keys_for_app_store_connect_api

**Comment by niico100** [comment]
This has now been moved, I've spent days trying to find Issuer ID now, every link I've found online is wrong, anyone know how to find it now?

So you have to generate a Team Key not an Individual Key to get Issuer ID? I'm just a solo dev though.
